ecs-agent/data/client.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package data
+
+import (
+	"encoding/json"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/modeltransformer"
+
+	"github.com/pkg/errors"
+	bolt "go.etcd.io/bbolt"
+)
+
+type NetworkDataClient interface {
+}
+
+type Client struct {
+	Accessor    DBAccessor
```

Comment:
What is the benefit of having `DBAccessor`? Can we make `PutObject`/`GetObject` directly methods of `Client`?

Reply:
My initial thought is to separate the direct DB operations from the interfaces that agent will call like `SaveTask` to prevent agent direct use. Originally those methods are not exported, I was thinking of honoring this tradition. Do you feel this is reasonable?